### PR TITLE
feat: Event Stream refactor

### PR DESCRIPTION
**Pull request by cynicaljoy**
ENG-6824

### Description
Rename the Event Stream resources

### Motivation and context
Consistent naming between the Features and the drivers 

### How was the change tested?
Unit/Integration tests

### Change types

* - [ ] Bug fix (non-breaking change that fixes an issue)
* - [x] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [x] My code follows the code style of this project.
* - [x] My change requires a change to Fauna documentation.
* - [x] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -338,11 +338,10 @@ func main() {
 }
 ```
 
-In query results, the driver represents stream tokens as `fauna.Stream`
+In query results, the driver represents stream tokens as `fauna.EventSource`
```
Comment (jrodewig) [inline README.md:341]:
```suggestion
In query results, the driver represents event sources  as `fauna.EventSource`
```

File: client_example_test.go [hunk 2]
```diff
@@ -342,11 +342,13 @@ func ExampleClient_Stream() {
...
 	streamQuery, _ := fauna.FQL(`StreamingSandbox.all().toStream()`, nil)
-	events, err := client.Stream(streamQuery)
+	events, err := client.StreamFromQuery(streamQuery, nil)
 	if err != nil {
 		log.Fatalf("failed to subscribe to the stream value: %s", err)
 	}
-	defer events.Close()
+	defer func() {
+		_ = events.Close()
+	}()
```
Comment (pnwpedro) [inline client_example_test.go:351]: I don't see the motivation for wrapping this.

File: client_example_test.go [hunk 3]
```diff
@@ -417,17 +419,19 @@ func ExampleClient_Subscribe() {
...
 	// initiate the stream subscription
-	events, err := client.Subscribe(stream)
+	events, err := client.Stream(stream)
 	if err != nil {
 		log.Fatalf("failed to subscribe to the stream value: %s", err)
 	}
-	defer events.Close()
+	defer func() {
+		_ = events.Close()
+	}()
```
Comment (pnwpedro) [inline client_example_test.go:434]: Same comment

**Review by pnwpedro — APPROVED** [review]
This lgtm, but don't we have to increment to v3 everywhere before we make these changes? https://github.com/search?q=repo:fauna/fauna-go%20v2&type=code

File: client_example_test.go (hunk 3 above)
Comment (cynicaljoy) [reply client_example_test.go:434]: to silence linter warnings about unhandled errors

File: client_example_test.go (hunk 2 above)
Comment (cynicaljoy) [reply client_example_test.go:351]: to silence linter warnings about unhandled errors

**Comment by cynicaljoy** [comment]
> This lgtm, but don't we have to increment to v3 everywhere before we make these changes? https://github.com/search?q=repo:fauna/fauna-go%20v2&type=code

my plan was to merge this to main and then version bump in a separate PR. do you think we should version bump within this one?

**Comment by pnwpedro** [comment]
> my plan was to merge this to main and then version bump in a separate PR. do you think we should version bump within this one?
yeah, i was thinking we'd move to v3 before. we can always cherry pick something to v2 and adjust the pipeline to deploy it if needed.

File: client_example_test.go (hunk 2 above)
Comment (pnwpedro) [reply client_example_test.go:351]: mmk, i'd rather either let the linter warn us, or actually do something with the error (like log it). but dropping it just to silence the fact that it's unhandled feels wrong.

**Review by jrodewig — APPROVED** [review]
LGTM. I pushed some updates to the README as a few things were wonky. Sorry for the noise there.

**Review by erickpintor — APPROVED** [review]
LGTM

**Pull request merged**